src/main/java/htsjdk/samtools/reference/AbstractIndexedFastaSequenceFile.java

```diff
@@ -146,6 +147,41 @@ protected static void sanityCheckDictionaryAgainstIndex(final String fastaFile,
         }
     }
 
+    /** Do some basic checking to make sure the fasta and the index match.
+     * <p>
+     * checks that the length of the fasta file is at least as long as the index proclaims
+     * and that beyond the last position references in the index there is only one line followed by whitespaces
+     *
+     * @param fastaFile Used for error reporting only.
+     * @param index index file to check against the dictionary.
+     */
+    public static void sanityCheckFastaAgainstIndex(final String fastaFile,
+                                                    final FastaSequenceIndex FastaSequenceIndex) {
+
+
+        final Iterator<FastaSequenceIndexEntry> iterator = FastaSequenceIndex.iterator();
+        FastaSequenceIndexEntry fastaSequenceIndex = null;
+        while (iterator.hasNext()) {
+            fastaSequenceIndex = iterator.next();
+        }
+        assert fastaSequenceIndex != null;
+        final long lastSequenceLength = fastaSequenceIndex.getSize();
+        final long lastSequenceStart = fastaSequenceIndex.getLocation();
+        final long lastSequenceEnd = lastSequenceStart + fastaSequenceIndex.getOffset(lastSequenceLength);
+
+        final long fastaLength = new File(fastaFile).length();
+
+        //Question: should we worry about files with lots of whitespace in their end?
+        if (lastSequenceEnd > fastaLength) {
+            throw new IllegalArgumentException("The fasta file is shorter (%d) than its index claims (%d). Please reindex the fasta.".formatted(fastaLength, lastSequenceEnd));
+        }
+        // not sure why need to add 1 here.
+        if (lastSequenceEnd + fastaSequenceIndex.getTerminatorLength() + 1 < fastaLength) {
```

Comment:
Honestly I wonder if we should allow for more than 1?  What if there are a handful of blank lines at the end?  Maybe we could try a nested solution, that _if_ the file is longer than you have here ... then we read the content after the last base and ensure it's all whitespace?